### PR TITLE
Network statistics documentation enhancements

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/reference/networking/net_stats.rst [hunk 1]
```diff
@@ -6,6 +6,25 @@ Network Statistics
...
+
+By default system collects network statistics per network interface. This
+can be controlled by :option:`CONFIG_NET_STATISTICS_PER_INTERFACE` option.
+
+The :option:`CONFIG_NET_STATISTICS_USER_API` option can be set if the
+application wants to collect statistics for further processing. The network
+management interface API is used for that. See :ref:`net_mgmt_interface` for
+details.
+
+If the :option:`CONFIG_NET_STATISTICS_ETHERNET_VENDOR` option is set, then
```
Comment (pfalcon) [inline doc/reference/networking/net_stats.rst]: Thanks for all the docs improvements. But why do we jump to CONFIG_NET_STATISTICS_ETHERNET_VENDOR here? Please first say that CONFIG_NET_STATISTICS_ETHERNET exists for generic Ethernet stats, and beyond that,  there's CONFIG_NET_STATISTICS_ETHERNET_VENDOR for additional stuff.

File: doc/reference/networking/net_stats.rst [hunk 2]
```diff
@@ -6,6 +6,25 @@ Network Statistics
 Overview
 ********
 
+Network statistics is collected if :option:`CONFIG_NET_STATISTICS` is set.
```
Comment (dbkinder) [inline doc/reference/networking/net_stats.rst]:
```suggestion
Network statistics are collected if :option:`CONFIG_NET_STATISTICS` is set.
```

File: doc/reference/networking/net_stats.rst [hunk 3]
```diff
@@ -6,6 +6,25 @@ Network Statistics
...
+Individual component statistics like for IPv4 or IPv6, can be turned off
```
Comment (dbkinder) [inline doc/reference/networking/net_stats.rst]:
```suggestion
Individual component statistics for IPv4 or IPv6 can be turned off
```

File: doc/reference/networking/net_stats.rst [hunk 4]
```diff
@@ -6,6 +6,25 @@ Network Statistics
...
+if those statistics are not needed. See various options in
+:file:`subsys/net/ip/Kconfig.stats` file for details.
+
+By default system collects network statistics per network interface. This
```
Comment (dbkinder) [inline doc/reference/networking/net_stats.rst]:
```suggestion
By default, the system collects network statistics per network interface. This
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/reference/networking/net_stats.rst (hunk 1 above)
Comment (jukkar) [reply doc/reference/networking/net_stats.rst]: I will add a comment about generic Ethernet stats.

**Comment by jukkar** [comment]
Updated according to comments.

**Review by pfalcon — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]

**Pull request merged**